src/unitxt/llm_as_judge.py

```diff
@@ -209,5 +242,73 @@ def compute(
                     "judge_raw_input": instance["source"],
                 }
             results.append(result)
+        return results
+
+    def prepare_instances(self, references, predictions, task_data):
+        input_instances = self._get_input_instances(task_data)
+        return self._get_instance_for_judge_model(
+            input_instances, predictions, references
+        )
+
+
+class TaskBasedLLMasJudge(LLMAsJudgeBase):
```

Comment:
why not have this class as the base class?

Reply:
and the original llm as judge should be implemented using this class as base

Reply:
@OfirArviv - let's discuss offline